ui/src/pages/Access/Roles.tsx

```diff
@@ -120,6 +130,16 @@ export default function Roles({ allRoles, refreshRoles }: RolesProps) {
     setModalVisible(true);
   };
 
+  const existingRole = useMemo(() => {
```

Comment:
Why not pass the `existingRole` from `Access/index` to `Access/Roles`? Actually you've calculated one in parent component.

Reply:
Done.